tests/public-dns-tests/src/macros.rs

```diff
@@ -0,0 +1,31 @@
+macro_rules! tests {
```

Comment:
I'd probably favor having a little boilerplate over all the macros. At the very least, there could be some docstrings that make it clearer what these ought to expand to, and why we need a macro here.